lib/Dialect/RTL/Ops.cpp

```diff
@@ -692,16 +690,13 @@ void AddOp::getCanonicalizationPatterns(OwningRewritePatternList &results,
       }
 
       auto mulOp = inputs[size - 1].getDefiningOp<rtl::MulOp>();
-      APInt multiplier;
-
       // add(..., x, mul(x, c)) -> add(..., mul(x, c + 1))
       if (mulOp && mulOp.inputs().size() == 2 &&
           mulOp.inputs()[0] == inputs[size - 2] &&
-          matchPattern(mulOp.inputs()[1], m_RConstant(multiplier))) {
-
-        APInt one(/*numBits=*/multiplier.getBitWidth(), 1, /*isSigned=*/false);
+          matchPattern(mulOp.inputs()[1], m_RConstant(value))) {
 
-        auto rhs = rewriter.create<ConstantOp>(op.getLoc(), multiplier + one);
+        APInt one(/*numBits=*/value.getBitWidth(), 1, /*isSigned=*/false);
+        auto rhs = rewriter.create<ConstantOp>(op.getLoc(), value + one);
         std::array<Value, 2> factors = {mulOp.inputs()[0], rhs};
```

Comment:
std::array is good, thanks!  You can also use a C array and wrap with an `ArrayRef(factors)` when you pass it, but std::array is better in this case.  Thanks!